### DISCUSSION TITLE
Bayesian optimization hyperparameters tuning

### DISCUSSION THREAD

**Discussion by Yassine97HF** [alan-turing-institute/sktime#3359] [Ideas]
Hello everyone!

i want to ask you if `sktime` supports Bayesian search for  hyperparameters tuning , there is a possibility to use  `hyperopt` by exemple with  sktime's pipelines ?

Thanks.

**Comment by fkiraly** [comment]
Not yet, but I think that's a great idea!

The generalization of that might be abstracting the optimizer and allowing it to slot in with a generic parameter search algorithm.

**Comment by BradKML** [comment]
Not just HyperOpt but `optuna` and others too. Bayesian and Bandit-like algorithms seems promising compared to Random Search or Grid Search (dunno about "Population Based Training" tho).

**Reply by BradKML** [reply]
Currently only these two exists:
- https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingRandomizedSearchCV.html
- https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html

**Comment by AugustComte** [comment]
Hi there, I was wondering if this had been scheduled in? And if so how far away might it be from release. I'm certain its not trivial, but it would be awesome to have the option for optuna for hyperparameter tuning.

**Comment by ilkersigirci** [comment]
I think sktime can follow [tune-sklearn](https://github.com/ray-project/tune-sklearn#tunegridsearchcv) way, which is drop in replacement for sklearn's GridSearch with cutting edge search methods.

**Comment by fkiraly** [comment]
Yes, I was thinking about planning this and/or showcasing this as a potential project for the summer programme, GSoC.

I have opened a good first issue which summarizes the discussion: https://github.com/sktime/sktime/issues/4188

**Reply by fkiraly** [reply]
also, if anyone here is interested doing this: let me know, I can draw up a design how to make this a "slot in" mechanism
